File: src/main/java/com/qlik/demo/java/model/BookingDocument.java
package com.qlik.demo.java.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.OffsetDateTime;
import java.util.UUID;

@Document(collection = "bookings")
public record BookingDocument(
        @Id String id,
        UUID userId,
        String destination,
        OffsetDateTime bookingDate
) {
}

File: src/main/kotlin/com/qlik/demo/kotlin/model/BookingDocument.kt
package com.qlik.demo.kotlin.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.OffsetDateTime
import java.util.*

@Document(collection = "bookings")
data class BookingDocument(
    @Id val id: String,
    val userId: UUID,
    val destination: String,
    val bookingDate: OffsetDateTime
)
